src/main/java/seedu/address/logic/ParserManager.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic;
+
+import seedu.address.logic.parser.FeatureParser;
+import seedu.address.logic.parser.ModuleListParser;
+import seedu.address.logic.parser.TodoListParser;
+import seedu.address.logic.parser.contactlistparsers.ContactListParser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents the manager in charge of overseeing all the medium feature parsers.
+ */
+public class ParserManager {
+    private final ModuleListParser moduleListParser;
+    private final ContactListParser contactListParser;
+    private final TodoListParser todoListParser;
+
+    /**
+     * Creates a Container the holds all the medium parsers.
+     * @param moduleListParser parses module related commands.
+     * @param todoListParser parses task related commands.
+     * @param contactListParser parses contact related commands.
+     */
+    public ParserManager(ModuleListParser moduleListParser,
+                         TodoListParser todoListParser,
+                         ContactListParser contactListParser) {
+        this.contactListParser = contactListParser;
+        this.todoListParser = todoListParser;
+        this.moduleListParser = moduleListParser;
+    }
+
+    /**
+     * Selects which medium parser to be used based on the command word.
+     * @param commandWord the command word of the user input.
+     * @return FeatureParser a medium parser.
+     * @throws ParseException when the command word is invalid.
+     */
+    public FeatureParser select(String commandWord) throws ParseException {
+        if (commandWord.contains("module")) {
```

Comment:
For this method, we should avoid using the contains() method. Consider a scenario where the user inputs "module1" as the first input word by accident. Then "module1" would be accepted as a valid command word. This can give rise to bugs and complications. We could consider using .equals() instead.

Reply:
but if we use .equals(), we need to check every single existing command, which is alot.

Reply:
@jonasngs I think it should be fine since this is just the first layer for checking the commands. It will be checked again in each of the FeatureParser e.g. even though "addmodule1" may pass the ParserManager, the method parseCommand in ModuleListParser will throw an exception anyway.

In addition, I think we can consider using static HashSet for checking the type of the command. It will be accurate (less prone to bug) and also fast at the same time (no need to check for every existing command), but of course it will consume more memory (which one is our priority though?) 😃

Reply:
Yes, that makes sense, we could consider that.